src/player/brains/cPlayerBrainSandworm.cpp

```diff
@@ -38,16 +38,21 @@ void cPlayerBrainSandworm::think()
 
 void cPlayerBrainSandworm::findRandomValidLocationToMoveToAndGoThere(cUnit &pSandWorm) const
 {
-    // find new spot to go to
+    bool foundPlaceToMoveTo = false;
     for (int iTries = 0; iTries < 5; iTries++) {
         int iMoveTo = global_map.getRandomCellWithinMapWithSafeDistanceFromBorder(2);
 
         if (global_map.isCellPassableForWorm(iMoveTo)) {
             pSandWorm.move_to(iMoveTo);
```

Comment:
Extract so that "find place to move to" returns either -1 or a cell. Then there is no need for boolean and it further simplifies logic